### PR TITLE
build: update to klog v2.70.0

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

The main practical advantage is that klog.Fatal no longer dumps the backtrace
of all goroutines.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: vendor/k8s.io/klog/v2/internal/serialize/keyvalues.go [hunk 1]
```diff
@@ -44,53 +46,49 @@ func WithValues(oldKV, newKV []interface{}) []interface{} {
 	return kv
 }
 
-// TrimDuplicates deduplicates elements provided in multiple key/value tuple
-// slices, whilst maintaining the distinction between where the items are
-// contained.
-func TrimDuplicates(kvLists ...[]interface{}) [][]interface{} {
```
Comment (liggitt) [inline vendor/k8s.io/klog/v2/internal/serialize/keyvalues.go:50]: did this method go away? how did that pass the klog apidiff compatibility checks?

Comment (liggitt) [reply vendor/k8s.io/klog/v2/internal/serialize/keyvalues.go:50]: edit: oh... it's under internal/..., nevermind

File: vendor/k8s.io/klog/v2/k8s_references.go [hunk 2]
```diff
@@ -77,6 +77,8 @@ func KRef(namespace, name string) ObjectRef {
 }
 
 // KObjs returns slice of ObjectRef from an slice of ObjectMeta
+//
+// DEPRECATED: Use KObjSlice instead, it has better performance.
```
Comment (liggitt) [inline vendor/k8s.io/klog/v2/k8s_references.go:81]: is there an issue tracking switching use to this in a follow up?

File: vendor/k8s.io/klog/v2/internal/serialize/keyvalues.go (hunk 1 above)
Comment (pohly) [reply vendor/k8s.io/klog/v2/internal/serialize/keyvalues.go:50]: I had to get apidiff fixed first to have it accept this change without complaining, but yes, this is exactly the reason :smile:

File: vendor/k8s.io/klog/v2/k8s_references.go (hunk 2 above)
Comment (pohly) [reply vendor/k8s.io/klog/v2/k8s_references.go:81]: There is now: https://github.com/kubernetes/kubernetes/issues/110737

**Pull request merged**